src/commitlint/linter/_linter.py

```diff
@@ -0,0 +1,69 @@
+"""
+This module provides detailed functionality for linting commit messages according
+to conventional commit standards.
+"""
+
+from typing import List, Tuple
+
+from .quick import quick_lint_commit_message
+from .utils import is_ignored, remove_comments
+from .validators import HeaderLengthValidator, PatternValidator
+
+
+def lint_commit_message(
+    commit_message: str, quick: bool = False
+) -> Tuple[bool, List[str]]:
+    """
+    Lints a commit message.
+
+    Args:
+        commit_message (str): The commit message to be linted.
+        quick (bool, optional): Whether to perform a quick linting (default is True).
+
+    Returns:
+        Tuple[bool, List[str]]: Returns success as a first element and list of errors
+            on the second elements. If success is true, errors will be empty.
+    """
+
+    # perform processing and pre checks
+    # removing unnecessary commit comments
+    commit_message = remove_comments(commit_message)
+
+    # checking if commit message should be ignored
+    if is_ignored(commit_message):
+        return True, []
+
+    # for quick check
+    if quick:
+        return quick_lint_commit_message(commit_message)
+
+    return _detailed_lint_commit_message(commit_message)
+
+
+def _detailed_lint_commit_message(commit_message: str) -> Tuple[bool, List[str]]:
```

Comment:
Like quick.py, why not separate it into a detailed.py file?

Reply:
@subashcs, My thought was to make detailed linting the primary linting process and thus I have added it to the `_linter.py` file.

We can also separate it into `detailed.py`. For that, we should keep the `lint_commit_message` function in __init__.py, which calls quick and detailed linting from their respective files (quick.py and detailed.py).

@subashcs, What are your further suggestions?

Reply:
The second option sounds good.

Reply:
@subashcs, following your other comment about using the validators generically, there seems to be no need to create different files for detailed and quick linting. Please review the PR for the changes.